src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py

```diff
@@ -243,33 +240,20 @@ def passesSiteRestriction(self, site):
 
         return True
 
-    def intersectionWithEmptySet(self, a, b):
-        """
-        interaction of 2 sets but if one of the set is empty returns union
-        """
-        if len(a) == 0 or len(b) == 0:
-            return a | b
-        else:
-            return a & b
-
     def possibleSites(self):
```

Comment:
Yes, you are right it might be better to make SiteWhitelist mandatory. Let's wait what Jean-Roch says.